src/bridge/BitcoinTx.sol

```diff
@@ -280,4 +291,32 @@ library BitcoinTx {
 
         revert("No output found for scriptPubKey");
     }
+
+    function ensureTxInputSpendsUtxo(bytes memory _vin, BitcoinTx.UTXO memory utxo) internal pure {
+        uint256 _varIntDataLen;
+        uint256 _nIns;
+
+        (_varIntDataLen, _nIns) = BTCUtils.parseVarInt(_vin);
+        require(_varIntDataLen != BTCUtils.ERR_BAD_ARG, "Read overrun during VarInt parsing");
+
+        uint256 _len = 0;
+        uint256 _offset = 1 + _varIntDataLen;
+
+        for (uint256 _i = 0; _i <= _nIns; _i++) {
+            _len = BTCUtils.determineInputLengthAt(_vin, _offset);
+            require(_len != BTCUtils.ERR_BAD_ARG, "Bad VarInt in scriptSig");
+            _offset = _offset + _len;
```

Comment:
I think you need to do this at the end of the for loop (after extracting the input)

Reply:
Done, but after the loop ends again need to get the input and check utxo right?